### ISSUE TITLE
bun crashed "Segmentation fault"

### ISSUE THREAD

**Issue by scffs** [oven-sh/bun#11944]
### How can we reproduce the crash?

I wrote a simple code that only make requests to server in a while. It is possible that the crash is in PC lock mode (Control-Command-Q)

### Relevant log output

```shell
Bun v1.1.13 (bd6a6051) macOS Silicon
Args: "bun" "run" "--hot" "src/main.js"
Features: jsc Bun.stdin(2) fetch(9964) transpiler_cache(8) tsconfig
Builtins: "bun:main" "node:assert" "node:buffer" "node:crypto" "node:dns" "node:events" "node:fs" "node:fs/promises" "node:http" "node:net" "node:os" "node:process" "node:stream" "node:string_decoder" "node:timers" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib"
Elapsed: 7393199ms | User: 94868ms | Sys: 32311ms
RSS: 0.25GB | Peak: 0.27GB | Commit: 1.07GB | Faults: 1753

panic: Segmentation fault at address 0x48C00F23C80
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```

### Stack Trace (bun.report)

Bun v1.1.13 ([`bd6a605`](<https://github.com/oven-sh/bun/tree/bd6a6051203ff577240615ff4e242270035a36b7>)) on macos aarch64 [RunCommand]

Segmentation fault at address 0x48C00F23C80

- *2 unknown/js code*
- `src.http.onWritable__anon_482057`
- `src.deps.uws.NewSocketHandler`
- `us_loop_run_bun_tick`
- `src.http.HTTPThread.processEvents`
- `Thread.PosixThreadImpl.spawn__anon_292305.Instance.entryFn`
- *2 unknown/js code*

in the 1.1.12 version of bun this code works fine (all day without crashing)

**Comment by Jarred-Sumner** [comment]
Any chance you could provide some code for us to reproduce the issue? The crash is saying that it is something to do with the HTTP client. But it's hard to assess further than that.

**Comment by scffs** [comment]
> Any chance you could provide some code for us to reproduce the issue? The crash is saying that it is something to do with the HTTP client. But it's hard to assess further than that.

yes sure

```js
import { connect } from 'mongoose'

const delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms))

const connectToDatabase = async () => {
  try {
    await connect('mongodb://localhost:27017/test')
  } catch (error) {
    process.exit(1)
  }
}

export const getSmth = async () =>
  fetcher('https://jsonplaceholder.typicode.com/posts')

const fetcher = async (url, body = '{}') => {
  let items

  const response = await fetch(url, {
    method: 'POST',
    headers: {
      Authorization: '123',
      'Content-Type': 'application/json'
    },
    body
  }).catch((error) => {
    console.error(`Error (${new Date().toISOString()}): ${error}`)
  })

  try {
    const r = await response.json()

    // IDK why there is no return but it works
    if (response.status !== 200) {
      console.error(r)
    }

    console.log('r', r)

    items = r
  } catch (err) {
    return null
  }

  await delay(450)
  return items
}

// a simplified version of my code
async function main() {
  // could comment this I guess
  await connectToDatabase()

  // I'm getting some data to display it in the console
  const someData = await getSmth()

  while (true) {
    const newInterval = 1000

    try {
      await getSmth()
    } catch (error) {
      console.error(error)
    }

    // in my case, I get an array of some elements and in a loop I run a different query for each of these elements
    for (let i = 0; i < 10; i++) {
      await getSmth()
    }

    await getSmth()
    await delay(newInterval)
  }
}

main()
```

**Comment by scffs** [comment]
@Jarred-Sumner, hi again

According to my observations, the script really crashes after the screen is locked or during somehow 

![image](https://github.com/oven-sh/bun/assets/67464545/704ed5dc-1dfc-46ef-bee1-751933c8367a)

by the way, I have 2 identical scripts running (let's imagine that I have 2 identical projects in different folders and work independently of each other) and as a rule only 1 crashes

**Comment by cirospaciari** [comment]
Can you still reproduces it on `v0.0.20`? This looks like a issue that is already solved

**Comment by scffs** [comment]
> Can you still reproduces it on `v0.0.20`? This looks like a issue that is already solved

I'll check this today

**Comment by scffs** [comment]
looks like fixed

but can you please provide any information? like PR that fixed this bug or whatever

@cirospaciari

**Comment by scffs** [comment]
<img width="1007" alt="image" src="https://github.com/user-attachments/assets/f064ec75-c679-4c28-9a8d-8e18e98225c2">

I got this error again xd @cirospaciari

**Comment by cirospaciari** [comment]
We added some fixes into sockets, can you try bun upgrade --canary and check if is still happening?

**Comment by scffs** [comment]
fixed at 1.1.24
